WireDomain/Sources/WireDomain/UseCases/OneOnOneResolverUseCase.swift

```diff
@@ -0,0 +1,307 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import CoreData
+import WireAPI
+import WireDataModel
+
+// sourcery: AutoMockable
+/// Resolves 1:1 conversations
+public protocol OneOnOneResolverUseCaseProtocol {
+    func invoke() async throws
+}
+
+public struct OneOnOneResolverUseCase: OneOnOneResolverUseCaseProtocol {
```

Comment:
Following our discussion [here](https://github.com/wireapp/wire-ios/pull/1994), I will rename this to `OneOnOneResolver` and remove public access for that resolver (since it is only meant to be used internally by WireDomain)

Reply:
done in https://github.com/wireapp/wire-ios/pull/1999/commits/79ab5abd99f212af6d35c47b3b6a72f9ff8bde19